src/libstd/macros.rs

```diff
@@ -410,7 +410,7 @@ mod builtin {
     ///
     /// [`panic!`]: ../std/macro.panic.html
     #[stable(feature = "compile_error_macro", since = "1.20.0")]
-    #[rustc_doc_only_macro]
+    #[rustc_builtin_macro]
     macro_rules! compile_error {
```

Comment:
Oh, these are unchanged. What needs to happen before these can just be reexports of the `libcore` ones?

Reply:
`rustc_builtin_macro` stubs acting like the real macros.

https://github.com/rust-lang/rust/pull/62086 removes the duplicate macros from libstd, but also `pub use`s their libcore versions from libstd's root (for documentation). 
...And stub macros `pub use`d in libstd root break all the dependent crates that inject `#[macro_use] extern crate std` during rustdoc build.

Alternatively, some `doc` attribute for selectively documenting private items would also solve the problem (`pub use` could be replaced with `#[doc(force)] use` then).